### PR TITLE
Autism Fort Patch - Emitters and Tesla coils can now be used for DEFENDING YOUR TURF!

### PR DESCRIPTION
**Pull request by silicons**
Gives emitters and tesla coils a wire that when pulsed will make emitters fire a shot and tesla coils to send out lightning. Both have cooldowns reliant on the stock part rating inside them!
:cl:
rscadd: Emitters and Tesla Coils now have activation wires!
rscadd: Emitters will shoot out an emitter bolt when pulsed, regardless of it is on.
rscadd: Tesla coils will shoot lightning when pulsed, if it is connected to a cable that has power.
/:cl:

### CODE REVIEW

**Comment by kilkun** [comment]
interesting idea

**Comment by jughu** [comment]
But mekhi before we run in any intresting bugs... does it converse power?

**Comment by jughu** [comment]
Because if it dont i tottaly can make a paper plane V2  with a prox/signaller assembly and a signaller and a monkey disposal system

**Comment by kilkun** [comment]
stfu dude dont tell the coders how to fix the game

**Comment by kilkun** [comment]
do you hate fun???

**Comment by PKPenguin321** [comment]
This is the opposite of what a tesla coil does :thinking:

File: code/modules/power/tesla/coil.dm [hunk 1]
```diff
@@ -66,6 +75,19 @@
...
+		return FALSE
+	last_zap = world.time
+	var/coeff = (12 - (input_power_multiplier * 3))
+	coeff = max(coeff, 1)
+	var/shock_coeff = (5 - input_power_multiplier)
+	shock_coeff = max(shock_coeff, 1)
+	var/power = (powernet.avail/coeff)
+	add_load(power)
+	playsound(src.loc, 'sound/magic/LightningShock.ogg', 100, 1, extrarange = 5)
+	tesla_zap(src, 10, power/shock_coeff)
```
Comment (Core0verload) [inline code/modules/power/tesla/coil.dm]: Tesla coil would create INFINITE POWER again, like the paper engine. Make sure to fix it, then double-check the fix.

File: code/modules/power/singularity/emitter.dm [hunk 2]
```diff
@@ -168,40 +168,45 @@
...
+			A.xo = 0
+		if(EAST)
+			A.yo = 0
+			A.xo = 20
+		if(WEST)
+			A.yo = 0
+			A.xo = -20
+		else // Any other
+			A.yo = -20
+			A.xo = 0
```
Comment (Core0verload) [inline code/modules/power/singularity/emitter.dm]: Looks like new shitcode, but it's a copypaste of old shitcode. How about fixing it anyway?

**Review by Core0verload — CHANGES_REQUESTED** [review]

File: code/modules/power/singularity/emitter.dm (hunk 2 above)
Comment (kevinz000) [inline code/modules/power/singularity/emitter.dm]: It's the same thing that I moved over for the proc.

File: code/modules/power/tesla/coil.dm (hunk 1 above)
Comment (kevinz000) [inline code/modules/power/tesla/coil.dm]: HMMMMMMMMMMMMMMMMMMMMMMMMMMM
How do I fix this.. Eh, I'l check my math later, see if I can balance it out.

**Comment by XDTM** [comment]
Hey can you instead make two tesla coils properly rigged form a wall of light like in dishonored?

File: code/modules/power/tesla/coil.dm (hunk 1 above)
Comment (optimumtact) [inline code/modules/power/tesla/coil.dm]: Just use more power than it gives dummy

Comment (Iamgoofball) [inline code/modules/power/tesla/coil.dm]: tesla coils already drop half the power

**Comment by Kromgar** [comment]
>Disintrigrating walls


Balanced

**Comment by jughu** [comment]
@kevinz000 also does the emitter pulse firing have a cooldown so i dont can create some super blob remover by spam pressing  a remote

**Comment by Iamgoofball** [comment]
@XDTM just use shield generators to do that

**Comment by ExcessiveUseOfCobblestone** [comment]
spectrum tower

**Comment by Anonmare** [comment]
@PKPenguin321 Do you mean in real life? Because I'm pretty sure that is how they work

**Comment by dannno** [comment]
sounds funny

**Comment by CPTANT** [comment]
emitters fire without even being bolted to the ground or being powered when pulsed.....

**Comment by PKPenguin321** [comment]
Mekhi code back at it again

**Pull request merged**